x-pack/plugins/code/public/sagas/repository.ts

```diff
@@ -125,13 +128,15 @@ function requestRepoInitCmd(repoUri: string) {
   });
 }
 function* handleGotoRepo(action: Action<string>) {
-  const repoUri = action.payload as string;
-  const repo = yield call(requestRepo, repoUri);
-  history.replace(`${repoUri}/tree/${repo.defaultBranch || 'master'}`);
-}
-
-function requestRepo(uri: string) {
-  return kfetch({ pathname: `/api/code/repo${uri}`, method: 'get' });
+  try {
+    const repoUri = action.payload as string;
+    yield put(loadRepo(repoUri));
+    const loadRepoSuccessAction = yield take(String(loadRepoSuccess));
+    history.replace(`/${repoUri}/tree/${loadRepoSuccessAction.payload.defaultBranch || 'master'}`);
+  } catch (e) {
+    history.replace(`/${action.payload}/tree/master}`);
```

Comment:
Can we comment here why we redirect to /tree/master anyway? this is very important here.